app/models/searchable_record.rb

```diff
@@ -0,0 +1,18 @@
+# The base class for all searchable models
+class SearchableRecord < ApplicationRecord
+  self.abstract_class = true
+
+  def self.searchable_attributes
+    []
+  end
+
+  def to_string
+    raise 'This method should be overriden to display a string when searching'
```

Comment:
Is there a cheap way to test that an exception is thrown when subclasses don't implement it?

Reply:
You mean a test that defines a new subclass of it and checks whether the exception occurs? or a test that create an instance of searchableRecord and checks that to_string throws error?

Reply:
I would mean the latter but I wasn't sure if you can even create an instance of it since it is abstract. If you can, then that would be a good test

Reply:
I added a test to check that using this class throws an error. So you need to subclass it

Reply:
```suggestion
  def to_string
    raise 'This method should be overridden to display a string when searching'
```